### PR TITLE
lldb_function_to_function_dbi: a hack to avoid crashing on C++ methods

### PR DESCRIPTION
**Pull request by yamt**
Also, print the function name on argument mismatch.

### CODE REVIEW

File: core/iwasm/compilation/debug/dwarf_extractor.cpp [hunk 1]
```diff
@@ -354,10 +354,27 @@ lldb_function_to_function_dbi(const AOTCompContext *comp_ctx,
...
-        LOG_ERROR(
-            "function args number dismatch!:value number=%d, function args=%d",
-            variable_list.GetSize(), num_function_args);
+        // A hack to detect C++ "this" pointer.
+        //
+        // REVISIT: is there a more reliable way?
+        // At the DWARF level, we can probably look at DW_AT_object_pointer
+        // and DW_AT_artificial. I'm not sure how it can be done via the
+        // LLDB API though.
+        if (num_function_args + 1 == variable_list.GetSize()) {
```
Comment (xujuntwt95329) [inline core/iwasm/compilation/debug/dwarf_extractor.cpp:365]: I also didn't find a way to get `DW_AT_object_pointer` from LLDB API.

This workaround seems work, but I have a concern on this condition: would it cause error if this is a vaargs function?
Of course this is not a new problem introduced by this PR, the L358 didn't consider vaargs function either. I'm not sure what will happen for vaargs functions here.

Comment (yamt) [reply core/iwasm/compilation/debug/dwarf_extractor.cpp:365]: my understanding is that, neither `num_function_args` or `variable_list.GetSize()` here include the hidden arg for stdarg.

Comment (xujuntwt95329) [reply core/iwasm/compilation/debug/dwarf_extractor.cpp:365]: Thanks

**Comment by xujuntwt95329** [comment]
LGTM

**Pull request merged**